core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -203,6 +203,35 @@ public DataFrameWriter write() {
         return new DataFrameWriter(defaultWriterRegistry, this);
     }
 
+    private void padColumnWithMissing(final Column<?> col, int newSize) {
+        while (col.size() < newSize) {
+            col.appendMissing();
+        }
+    }
+
+    /**
+     * Appends missing values to columns in the table and/or the Column argument,
+     * so their sizes end up being the same.
+     * @param cols
+     * @return this Table
+     */
+    public Table padColumnsWithMissing(final Column<?>... cols) {
```

Comment:
This seems like a strange method to me. It pads the given columns and columns in the table? Why? Presumably to enable adding those columns to the table, so why not do that here?

Reply:
I think there's also an open question about what should happen if the column padding makes the columns differ in size from the other columns in the table. Should that cause an exception to be thrown?

In general, it should not be possible to create a non-rectangular table, and adding columns that are the wrong size should cause an exception to be thrown. I think that means the only way to then make a non-rectangular table is to call append() on some but not all columns. 

To avoid problems, I might remove the columns argument from this method so the result was always a legal table. Again (though i haven't explicitly tried this) I would think that the safest way to add data to a table is to pad all the columns with missing and set the values in the new row(s). 

For me using column append() and forgetting to call appendMissing() on a column where there's no value to add is a source of frequent bugs.

Reply:
I think the tricky case is what if you add all equally sized columns to a table and then append to one columns and not the others. How would we catch that? Otherwise, I agree

Reply:
AFAICT, there is no reasonable way to prevent that, what I'm suggesting is "Don't do that". 
Do this instead:

```
t.padColumnsWithMissing(); // no params, so all columns are still the same size
t.stringColumn("foo").set(t.rowCount()-1, "my new value"); // no append so all columns are the same size

```

Reply:
@lwhite1 The method pads columns in the table too, so afterwards all the columns, both the ones already on the table and those given as arguments will be the same size. It makes sense to call this method without any arguments, to fix the table itself, and with additional arguments, to fix columns before adding them.
@benmccann Since the API is fluid, you'd do padColumnsWithMissing(cols).append(cols). We could also have both a non-args padColumnsWithMissing() to fix the table and a padWithMissingAndAppend(cols) to pad and append afterwards.

Reply:
> @lwhite1 The method pads columns in the table too, so afterwards all the columns, both the ones already on the table and those given as arguments will be the same size.

Nah, the table's columns are only the same size if you do it right. If you pad all the columns the result is always legal (assuming it was legal before you started).  If you specify the columns you can turn a legal table into an illegal one.